packages/@lwc/compiler/src/transformers/javascript.ts

```diff
@@ -34,10 +33,6 @@ export default function scriptTransform(
             plugins: [
                 [lwcClassTransformPlugin, { isExplicitImport, dynamicImports }],
                 [babelClassPropertiesPlugin, { loose: true }],
```

Comment:
I would go as far as to say we don't need this one neither if we accomodate my proposed changes :)

Reply:
Removing class field transformation break the existing decorators because we transpile using `loose` mode. It is something we need to address before even thinking about removing class field transformation today.

For context, we are attaching all the accessors for `@track` and `@api` directly on the class prototype. It works when Babel is transforming with `loose` mode since class fields are turned into property assignments in the constructor. This will not work without this transform as class fields act as property definitions and would shadow the accessors attached to the prototype. I wanted to make that change for a while now but I didn't take the time.

Reply:
Right but since we are doing the transform for decorators strictly before, wouldn't be ok to have the semantics of the remaining of the non-decorated fields with public props? 
This is not pressing anyway as it will take a while for the browser matrix to be far enough to even thinkking about doing this.

Reply:
No, it still won't with the current engine because of reactive properties. I ran the experiment locally this morning to confirm.

Similarly to `@track` properties, we set to assign a pair of getter/setter on the prototype. Those accessors are never called today if we let class fields as-is. One way to work around this would be to replace the class fields with accessors for each component instance right after the constructor. This will certainly introduce a performance overhead to the component construction, but I think it is worth the tradeoff.